src/header/internal.tsx

```diff
@@ -55,6 +57,7 @@ export default function InternalHeader({
         baseProps.className,
         styles[`root-variant-${variantOverride}`],
         isRefresh && styles.refresh,
+        hasToolbarHeader && styles['root-with-toolbar'],
```

Comment:
First design deviation for the toolbar feature flag. More to come in the future PRs